### DISCUSSION TITLE
Allow downloading libraries via Maven mirrors

### DISCUSSION THREAD

**Discussion by MiniDay** [PaperMC/Paper#11140] [Ideas]
### Is your feature request related to a problem?

Since 1.17, bukkit has added a plugin configuration option that allows downloading required libraries when the server starts, so plugin authors don't need to package them into the plugin file anymore, which is great.
However, the default Maven repository address is currently https://repo.maven.apache.org/maven2/
We have to consider that some users (like me) may not be able to access that address due to network conditions.
This will cause users (like me) to abandon some plug-ins because they cannot download dependencies.
So can we add a configuration option to allow users to define where their servers download dependencies from?

### Describe the solution you'd like.

Add the maven-mirror option in the configuration file (bukkit.yml or spigot.yml or config/paper-global.yml)

like this
```yaml
maven-mirror: https://repo.huaweicloud.com/repository/maven/
```

### Describe alternatives you've considered.

I haven't tried anything. My server cannot download the dependencies required by the plugin. There is nothing I can do about this problem except to file an issue here.

Before posting this issue, I read this: https://github.com/PaperMC/Paper/discussions/10763

I think it is not feasible to modify the dependency download address through  [Paper Plugins](https://docs.papermc.io/paper/dev/getting-started/paper-plugins), because users just want the plugins they find to work properly, and writing Java code for that is too painful

**Comment by toinouH** [comment]
Hello, I can add this to the conversation:

![image](https://github.com/user-attachments/assets/646b9882-9190-486b-8311-a031a305740b)
The mentionned pull request: https://github.com/PaperMC/Paper/pull/7177

**Comment by masmc05** [comment]
If you're not the author of the plugin, you may use [/etc/hosts](https://blog.purestorage.com/purely-educational/what-is-the-etc-hosts-file-in-linux/) to make that domain point to a local address which nginx will listen to, and make nginx act as a maven repository, where you put all the libraries your plugins may want to download. This way your plugin will work even if you don't have connection to the internet
